threads/main.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/c-bata/go-prompt"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/textileio/go-threads/api/client"
+	"github.com/textileio/go-threads/util"
+	"google.golang.org/grpc"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+var apiClient *client.Client
+var apiTimeout = time.Second * 5
+var streamTimeout = time.Second * 30
+var currentStore string
+
+type any map[string]interface{}
+
+var suggestions = []prompt.Suggest{
+	{
+		Text:        "use",
+		Description: "Switch the store.",
+	},
+	{
+		Text:        "exit",
+		Description: "Exit.",
+	},
+}
+
+var storeSuggestions = []prompt.Suggest{
+	{
+		Text:        "listen",
+		Description: "Stream all store updates.",
+	},
+	{
+		Text:        "getModel",
+		Description: "Get all models by model name.",
+	},
+}
+
+func completer(in prompt.Document) []prompt.Suggest {
+	w := in.GetWordBeforeCursor()
+	if w == "" {
+		return []prompt.Suggest{}
+	}
+	if currentStore != "" {
+		return prompt.FilterHasPrefix(append(suggestions, storeSuggestions...), w, true)
+	}
+	return prompt.FilterHasPrefix(suggestions, w, true)
+}
+
+func executor(in string) {
+	in = strings.TrimSpace(in)
+
+	blocks := strings.Split(in, " ")
+	switch blocks[0] {
+	case "use":
+		if len(blocks) < 2 {
+			fmt.Println("You must provide a store ID.")
+			return
+		}
+		currentStore = blocks[1]
+		fmt.Printf("Switched to %s\n", blocks[1])
+		return
+	case "listen":
+		listen(currentStore)
+		return
+	case "getModel":
+		if len(blocks) < 1 {
+			fmt.Println("You must provide a model name.")
+			return
+		}
+		getModel(currentStore, blocks[1], apiTimeout)
+		return
+	case "exit":
+		fmt.Println("Bye!")
+		os.Exit(0)
+	}
+	fmt.Println("Sorry, I don't understand.")
+}
+
+func getModel(id string, model string, maxAwaitTime time.Duration) {
+	ctx, cancel := context.WithTimeout(context.Background(), apiTimeout)
+	defer cancel()
+
+	rawResults, err := apiClient.ModelFind(ctx, id, model, nil, []*any{})
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	entities := rawResults.([]*any)
+	if len(entities) == 0 {
+		fmt.Printf("None found")
+		return
+	}
+	for _, el := range entities {
+		prettyPrint(el)
+	}
+}
+
+func listen(id string) {
+	channel, err := apiClient.Listen(context.Background(), id)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	for {
+		select {
+		case val, ok := <-channel:
+			if !ok {
+				return
+			} else {
+				if val.Err != nil {
+					fmt.Println(val.Err)
+					continue
+				}
+				obj := &any{}
+				if err := json.Unmarshal(val.Action.Entity, obj); err != nil {
+					fmt.Println("failed to unmarshal listen result")
+					continue
+				}
+				prettyPrint(obj)
+			}
+		case <-time.After(streamTimeout):
+			fmt.Println("timeout")
+			return
+		}
+	}
+}
+
+func main() {
+	addr := flag.String("addr", "/ip4/0.0.0.0/tcp/6006", "Threads APIs")
```

Comment:
qq—presumably the `addr` here will eventually be configurable (since you'll be able to connect either locally to the daemon or remotely to cloud.textile.io). is that right?

Reply:
This will set a default for the flag. So, you could do `threads -addr=/ip4/0.0.0.0/tcp/3000`.

Reply:
How about "Threads API address" for a description?